snapcraft/internal/repo/_deb.py

```diff
@@ -467,8 +471,9 @@ def _format_sources_list(sources_list, *,
             'arch': ' [arch={}]'.format(deb_arch) if foreign else '',
         })
     except KeyError as e:
-        raise errors.SnapcraftError(
-            'Sources list is missing the variable ${' + str(e) + '}')
+        raise ValueError(
+            'Cannot complete substitution in sources list: '
+            'unknown variable ${' + str(e.args[0]) + '} in template')
```

Comment:
Looks good, one nit: the format for a variable doesn't REQUIRE braces, it simply supports them. I wonder if simply using the variable name here would be better instead of assuming a specific format.

Reply:
I was kinda going by the fact that we use braces in our templates... but I thought of another option, I'm checking now which one was actually used to refer to it in the error message.